index.js

```diff
@@ -6,15 +7,258 @@ function ask(questionText) {
+let itemLookUp = {
+  'Energy Drink': energyDrink,
+  "coveralls": coveralls,
+  'ECM': ECM,
+  'Coffee Cup': coffeeCup,
+  'Journal': journal,
+  'notepad': notepad,
+  'Blueprint': blueprint,
+  'key': key,
+  "fuse": fuse
 }
+
+//------------------------------ movement to and from rooms -------------------------------
+
+async function moveToNewRoom(newRoom) {
+  let changeRoom = shipRooms[player.location].canChangeTo;
+  if (changeRoom.includes(newRoom)) {
+    if (player.inventory.includes('fuse')) {
+      hangarControlRoom['changePower']()
+      hangarControlRoom['changeLock']()
+    }
+    if (player.inventory.includes('key')) {
+      officerQuarters['changeLock']()
+    }
+    if (roomsLookUp[newRoom].locked) {
+      console.log('\n****This room is locked.**** ')
+    } else {
+      player.location = newRoom
+    }
+    if (player.inventory.includes('ECM') && player.location === 'hangar') {
+      console.log(`You rush back to your spacecraft in the hangar and immediately begin installation of your new ECM. With not a moment to spare you fix your drive, take off from the hangar, and begin your Quantum jump back home! \n\n!!!VICTORY!!!\n\nNow you're expected back at work on Monday...so "victory" is a subjective word `)
+      process.exit()
+    }
+  } else {
+    console.log('\nYou cant go that way! ')
+  }
+}
+
+//-----------------------------unlocking the engine block room--------------------------------
+async function passcodeEngineBlock() {
+  let passcodeQ = await ask('What is the Passcode ')
+  if (passcodeQ === '1223') {
+    engineBlock.locked = false
+    console.log('\n!!Access Granted!! ')
+  } else {
+    console.log('\n**Incorrect passcode** ')
+  }
+}
+//-----------------------------Turning on Power--------------------------------
+async function turnPowerOnInHangarControlRoom() {
+  if (hangarControlRoom.power === true) {
+    console.log('\n!!Access Granted!! ')
+  } else {
+    console.log('\nThe door\'s control panel seems fried; however, it seems like an easy fix... You\'ll just need to replace the fuse.')
+  }
+
+}
+// --------------------------- The game story code is below this line ---------------------------
+launch();
+
+async function launch() {
+  const welcomeMessage = `You don’t know how you survived your last encounter with space pirates.\nBut it seems luck was on your side\n...Well not entirely.\n\nYour ECM (electronic control module) was damaged in your last emergency jump drive\n\nWithout the ECM you’re unable to switch fuel types, making another quantum jump\nout of the Virgo system, back to Orion, your home system, impossible.\n\nAre you ready? `;
+  let answer = await ask(welcomeMessage);
+  if (answer !== 'yes') {
+    console.log(`*static radio* Incoming message from space pirates, 'I have you now!'...\nYou\'re disintegrated into space particles `)
+    process.exit()
+  } else {
+    console.log('\n\n!!!Alert: Your hydrogen fuel levels are critical. Immediate refueling required!!!');
+  }
```

Comment:
You can adjust these so that the more specific response (the user saying yes) is at the top, then you can add in a clause of "other responses" that would cause it to ask again, and a third clause saying that if it's a no it makes you lose the game. 

Doing this would give you better user input validation and a smoother play though

Reply:
This comment is also applicable for each of your logic statements from here down